pkg/replication/replicator.go

```diff
@@ -143,17 +150,24 @@ func (r replicator) initializeReplicationWithCheck(
 		concurrentRealityRequests = DefaultConcurrentReality
 	}
 
+	// NewTicker() requires the value passed to it to be positive.
+	if rateLimitInterval <= 0 {
+		rateLimitInterval = 1 * time.Nanosecond
```

Comment:
From testing on my laptop, the presence of a 1ns ticker brings one CPU to 50% utilization (https://play.golang.org/p/yzDjBIL6PI). **This is a dangerous default.**

Perhaps a 1s default timeout would be better, or you could selectively wait for the ticker if it exists.